src/main/java/org/asciidoc/intellij/ui/FormatterFactory.java

```diff
@@ -0,0 +1,21 @@
+package org.asciidoc.intellij.ui;
+
+import javax.swing.text.NumberFormatter;
+import java.text.NumberFormat;
+
+import static java.lang.Integer.MAX_VALUE;
+
+public final class FormatterFactory {
+
+  public static NumberFormatter createIntegerFormatter() {
```

Comment:
When editing the integer values in the form, I found that they are not consistent with other integer fields of IntelliJ.

What I observe in this implementation:
* when I type a "-" or characters, the implementation doesn't show them
* when I try to delete the last digit, the editor field doesn't allow for this
* when I type four digits, the thousands delimiter appears

What I see with other digit input fields in IntelliJ: 
* I can enter any value
* errors are highlighted with a red border
* no thousands separator

![digit](https://user-images.githubusercontent.com/3957921/94988818-52e2bf80-0570-11eb-9708-ab48a04d8a5b.gif)

Reply:
Yes I wasn't quite happy with this behaviour either.

After some digging, I found the `IntegerField` component which is exactly this field :+1: 
Using this field instead makes this `NumberFormatter` obsolete.

* Regarding the config, I like your idea of the `5px` fallback value, which could be set as the minimum value here too.
* Regarding the error behaviour,
** The simplest way would probably be to use an Optional.empty() and omit the width property.
** I could add a change listener (or look into focus listener) to evaluate the content and disable the `OKAction` of the dialog accordingly.

What's your take on this?

Reply:
Thank you for all the changes you implemented. I pushed a small change to instead of the 5px as a default value keep the empty value (for example when a user pastes an SVG).

Regarding the error behavior you describe: Using the default value of 5px in case of an invalid content would come as a surprise to the user as they have seen something different in the dialog.  

Disabling the OK button as long as invalid values are present sounds like a good idea to me. Maybe allowing an empty value (and then not adding a width) would be reasonable here, so users could remove the invalid width and hit OK without un-checking the box. 

Therefore I think adding a change listener or something similar is a good idea to proceed. 

DialogWrapper offers a lot of options here; there is a method doValidate() that you can override for a complex validation. And once you override postponeValidate() and return false, a cyclic validations runs while the user types. 

Please let me know if you have questions, or when there is a new commit to review.

Thanks again for looking into this!

Reply:
Sounds good to me.
Regarding the minimum width of `5px`, what do you think about dropping it entirely?
Reason is how the `IntegerField` works, you can allow an empty field with `widthInputField.setCanBeEmpty(true);` although the number retrieved via `field.getValue()` will still be the defined minimum.
This makes it hard to differentiate if the value 5 was set or if an invalid value was chosen.

I suggest initializing the `IntegerField` with `0` as the minimum integer and allow empty values.
So all invalid or empty values default to 0, which we than can omit and don't add to the options arguments as you suggested.

Regarding the wrapper validation, as I can tell `doValidate()` is only called once the confirm button of the dialog is already pressed. My though was to play it over the `IntegerField`s `addPropertyChangeListener` directly, but I will look into it.

Reply:
I follow your suggestion, let's drop the `5px` entirely.

You describe the default, that `doValidate()` is called once the confirm button has been pressed. But this default behavior changes once `postponeValidate()` is overwritten and returns `false`. 

I pushed a change to show the behavior. As the validation runs in cycles it's a brute force approach. Feel free to use a change listener instead. 

![validation](https://user-images.githubusercontent.com/3957921/95019279-be0dbe00-0664-11eb-9875-472feb99dde5.gif)

Reply:
Yes I'll change that, and also adjust the tool tip as we have the validation info now. 
Thanks for the help :+1: